src/View/Form/AbstractContext.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Cake\View\Form;
+
+use Cake\Network\Request;
+
+/**
+ * Provides common functionality for ContextInterface implementations.
+ */
+abstract class AbstractContext implements ContextInterface
+{
+    /**
+     * The request object.
+     *
+     * @var \Cake\Network\Request
+     */
+    protected $_request;
+
+    /**
+     * The request type of the attached form.
+     *
+     * @var string
+     */
+    protected $_requestType;
+
+    /**
+     * Constructor.
+     *
+     * @param \Cake\Network\Request $request The request object.
+     */
+    protected function __construct(Request $request)
+    {
+        $this->_request = $request;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setRequestType($requestType)
+    {
+        $this->_requestType = $requestType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function val($field)
```

Comment:
In reference to your example `val()` method I wouldn't want form fields to auto filled from querystring when `$this->_requestType` is POST/PUT. While it can help with PRG search form, it can cause unnecessary site effects in other cases. There can be times when someone is using querystring var and form field with same name which are not directly related.